locksmith/src/controllers/userController.ts

```diff
@@ -91,6 +100,45 @@ export const retrieveEncryptedPrivatekey = async (
   }
 }
 
+export const retrieveWaasUuid = async (
+  req: Request,
+  res: Response
+): Promise<any> => {
+  const apiKeyName = process.env.COINBASE_CLOUD_API_KEY_NAME
+  const privateKey = process.env.COINBASE_CLOUD_PRIVATE_KEY
+
+  const { emailAddress, selectedProvider } = req.params
+
+  let userUUID
+
+  const user = await UserOperations.findUserAccountByEmail(
+    req.params.emailAddress
+  )
+  userUUID = user?.id
+
+  // If no user is found, create
+  if (!user) {
+    if (!selectedProvider) return res.sendStatus(400)
```

Comment:
Please add a console log error here so we know _why_ he API is yielding an error. Also this should be a 500 not a 400